api.go

```diff
@@ -672,9 +685,17 @@ func (c Client) makeTargetURL(bucketName, objectName, bucketLocation string, que
 	host := c.endpointURL.Host
 	// For Amazon S3 endpoint, try to fetch location based endpoint.
 	if s3utils.IsAmazonEndpoint(c.endpointURL) {
-		// Fetch new host based on the bucket location.
-		host = getS3Endpoint(bucketLocation)
+		if c.isS3AccelerateEndpoint {
```

Comment:
To support S3 transfer acceleration shouldn't minio-go validate the conditions mentioned in the docs?
http://docs.aws.amazon.com/AmazonS3/latest/dev/transfer-acceleration.html#transfer-acceleration-requirements
e.g, this feature is applicable only on buckets where transfer acceleration is configured.